checkov/terraform/checks/resource/aws/LambdaEnvironmentCredentials.py

```diff
@@ -16,17 +16,19 @@ def __init__(self):
 
     def scan_resource_conf(self, conf):
         if 'environment' in conf.keys():
-            if 'variables' in conf['environment'][0]:
-                if type (conf['environment'][0]['variables']) is not list:
-                    conf['environment'][0]['variables'] = [conf['environment'][0]['variables']]
-                if type(conf['environment'][0]['variables'][0]) is dict:
-                    # variables can be a string, which in this case it points to a variable
-                    for values in list(conf['environment'][0]['variables'][0].values()):
-                        if type(values) is not list:
-                            values = [values]
-                        for value in values:
-                            if re.match(access_key_pattern, value) or re.match(secret_key_pattern, value):
-                                return CheckResult.FAILED
+            if isinstance(conf['environment'][0], dict):
+                if 'variables' in conf['environment'][0]:
+                    if type (conf['environment'][0]['variables']) is not list:
+                        conf['environment'][0]['variables'] = [conf['environment'][0]['variables']]
```

Comment:
@arielkru It seems like we're doing this quite a lot. Maybe it is time to extract to a method?